docs/prio-processor-guide.md

```diff
@@ -0,0 +1,371 @@
+        │   └── part-00001-6adba759-6e58-4092-8120-6331705e2e46.c000.json
+        └── batch_id=content.blocking_blocked_TESTONLY-1
+            ├── part-00002-6adba759-6e58-4092-8120-6331705e2e46.c000.json
+            └── part-00003-6adba759-6e58-4092-8120-6331705e2e46.c000.json
+```
+
+### Filesystem exchange
+
+The co-processors share data by using cloud storage. Each storage unit is
+separated by path hierarchy and permissions implemented by the filesystem. The
+path encodes various metadata.
+
+| Attribute       | Purpose                                                 |
+|-----------------|---------------------------------------------------------|
+| submission-date | The date of the batch processing submission.            |
+| server-id       | The recipient of the flattened and partitioned shares.  |
+| batch-id        | Used to identify the encoding and size of the data.     |
+| part-id         | Identify a partition in an ordered range of partitions. |
+
+The overall view of the path hierarchy.
+
+```bash
+filesystem
+├── server_a
+│   ├── intermediate
+│   │   ├── external
+│   │   │   ├── aggregate
+│   │   │   ├── verify1
+│   │   │   └── verify2
+│   │   └── internal
+│   │       ├── aggregate
+│   │       ├── verify1
+│   │       └── verify2
+│   ├── processed
+│   └── raw
+└── server_b
+    ├── intermediate
+    │   ├── external
+    │   │   ├── aggregate
+    │   │   ├── verify1
+    │   │   └── verify2
+    │   └── internal
+    │       ├── aggregate
+    │       ├── verify1
+    │       └── verify2
+    ├── processed
+    └── raw
+```
+
+The view of the paths when viewed from one project.
+
+```bash
+filesystem
+├── server_a
+│   ├── intermediate
+│   │   ├── external
+│   │   │   ├── aggregate
+│   │   │   ├── verify1
+│   │   │   └── verify2
+│   │   └── internal
+│   │       ├── aggregate
+│   │       ├── verify1
+│   │       └── verify2
+│   ├── processed
+│   └── raw
+└── server_b
+    └─── intermediate
+        └─── external
+            ├── aggregate
+            ├── verify1
+            └── verify2
+```
+
+### Triggering mechanism
+
+The processor is designed for ad-hoc usage that can be scheduled externally. The
+application starts up and waits for data in a specified location. When data is
+signaled in the receiving bucket, it processes the partition at a time, and
+writes it to the other server's receiving bucket. This process is repeated for
+all of the stages involved in aggregation: `verify1`, `verify2`, `aggregate`,
+and `publish`.
+
+Once all stages are complete, the processor will terminate and clear the state
+of the buckets.
+
+#### Retries and exponential back-off
+
+The following variables control exponential back-off:
+
+```bash
+RETRY_LIMIT
+RETRY_DELAY
+RETRY_BACKOFF_EXPONENT
+```
+
+### Configuring cloud storage
+
+There is support for s3 and gcs storage via
+[`gsutil`](https://cloud.google.com/storage/docs/interoperability).
+
```

Comment:
Probably want a little more detail about how to tell the tool about what buckets you have configured.